content/client-lib-development-guide/features.textile

```diff
@@ -378,14 +383,19 @@ h3(#realtime-channel). Channel
 * @(RTL11)@ If a channel enters the @DETACHED@ or @FAILED@ state, then all messages that are still queued for send on that channel should be deleted from the queue triggering a failure for the publish or presence methods invoked for those messages
 * @(RTL4)@ @Channel#attach@ function:
 ** @(RTL4a)@ If already @ATTACHED@ or @ATTACHING@ nothing is done
+** @(RTL4e)@ If the channel is in the @DETACHING@ state, it will result in an error, as the channel must complete the detach request before reattaching
+** @(RTL4g)@ If state is @FAILED@, the @attach@ request results in an error
 ** @(RTL4b)@ If the connection state is @CLOSED@, @CLOSING@, @SUSPENDED@ or @FAILED@, the @attach@ request results in an error
+** @(RTL4h)@ If the connection state is @INITIALIZED@, @CONNECTING@ or @DISCONNECTED@, do the operation once the connection state is @CONNECTED@
 ** @(RTL4c)@ Otherwise an @ATTACH@ ProtocolMessage is sent to the server, the state changes to @ATTACHING@ and the channel becomes @ATTACHED@ when the confirmation @ATTACHED@ ProtocolMessage is received
 ** @(RTL4f)@ Once an @ATTACH@ @ProtocolMessage@ is sent, if an @ATTACHED@ @ProtocolMessage@ is not received within the "default realtime request timeout":#defaults, the attach request should be treated as though it has failed
 ** @(RTL4d)@ If the language permits, a callback can be provided that is called when the channel is attached successfully or the attach fails and the @ErrorInfo@ error is passed as an argument to the callback
 ** @(RTL4e)@ If the user does not have sufficient permissions to attach to the channel, the channel state @FAILED@ is emitted with the error @ErrorInfo@. An error event is also emitted, and the @Channel#errorReason@ attribute of the @Channel@ is set to the error @ErrorInfo@ object
 * @(RTL5)@ @Channel#detach@ function:
 ** @(RTL5a)@ If state is @INITIALIZED@, @DETACHED@ or @DETACHING@ nothing is done
 ** @(RTL5b)@ If state is @FAILED@, the @detach@ request results in an error
+** @(RTL5g)@ If the connection state is @CLOSING@ or @FAILED@, the @detach@ request results in an error
```

Comment:
You could say that but, as the connection being CLOSED necessarily implies that the channel is DETACHED per [RTN18b](http://docs.ably.io/client-lib-development-guide/features/#RTN18b), this case is covered by RTL5a two lines above. The table also reflects that.